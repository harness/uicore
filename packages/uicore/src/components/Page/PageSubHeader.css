/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

.container {
  background: var(--white) !important;
  padding: var(--spacing-5) var(--spacing-xlarge) !important;
  border-bottom: 1px solid var(--grey-200);
  height: 65px;
}
