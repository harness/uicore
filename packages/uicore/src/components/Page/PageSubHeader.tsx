/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Layout } from '../../'
import React from 'react'
import cx from 'classnames'
import css from './PageSubHeader.css'

export interface PageSubHeaderProps {
  className?: string
}

/**
 * PageSubHeader implements a consistent sub header for a page in which content which is passed as children rendered
 *  with standard padding given by design team - top and bottom: 16px, left and right: 24px
 */
export const PageSubHeader: React.FC<PageSubHeaderProps> = ({ className, children }) => {
  return (
    <Layout.Horizontal flex className={cx(css.container, className)} data-testid="page-subheader">
      {children}
    </Layout.Horizontal>
  )
}
