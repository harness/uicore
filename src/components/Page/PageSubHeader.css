.container {
  background: var(--white) !important;
  padding: var(--spacing-5) var(--spacing-xlarge) !important;
  border-bottom: 1px solid var(--grey-200);
  height: 65px;
}
