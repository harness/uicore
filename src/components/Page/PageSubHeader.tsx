import { Layout } from '../../'
import React from 'react'
import cx from 'classnames'
import css from './PageSubHeader.css'

export interface PageSubHeaderProps {
  className?: string
}

/**
 * PageSubHeader implements a consistent sub header for a page in which content which is passed as children rendered
 *  with standard padding given by design team - top and bottom: 16px, left and right: 24px
 */
export const PageSubHeader: React.FC<PageSubHeaderProps> = ({ className, children }) => {
  return (
    <Layout.Horizontal flex className={cx(css.container, className)}>
      {children}
    </Layout.Horizontal>
  )
}
