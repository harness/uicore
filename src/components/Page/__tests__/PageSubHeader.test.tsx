import React from 'react'
import { noop } from 'lodash-es'
import { render } from '@testing-library/react'
import { Button } from '../../../'
import { PageSubHeader } from '../PageSubHeader'

describe('PageSubHeader test', () => {
  test('snapshot testing', () => {
    const { container } = render(
      <PageSubHeader>
        <Button intent="primary" data-testid="add-pipeline" text={'Add Pipeline'} onClick={noop} />
      </PageSubHeader>
    )
    expect(container).toMatchSnapshot()
  })
})
