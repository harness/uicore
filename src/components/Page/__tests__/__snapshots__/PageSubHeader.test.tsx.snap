// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSubHeader test snapshot testing 1`] = `
<div>
  <div
    class="horizontal layout-spacing-undefined main container flex flex-true"
  >
    <button
      class="bp3-button button font main intent intent-primary with-current-color"
      data-testid="add-pipeline"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        Add Pipeline
      </span>
    </button>
  </div>
</div>
`;
